block/manager.go

```diff
@@ -357,21 +487,13 @@ func (m *Manager) RetrieveLoop(ctx context.Context) {
 	for {
 		select {
 		case <-waitCh:
-			for {
```

Comment:
removing the inner-loop may slow down the block retrieval for both DA and blockstore. within a timer click, if the block is not found when the retrieve happens, but immediately afterwards if the block becomes available, you will still wait until the next timer tick. if the inner-loop exists, it will act as retry within the same timer click. what do you think?

Reply:
Yes I understand. Right now, without the inner loop, we check for new blocks every `BlockTime`. Re-adding an inner loop results in the inner for loop busy-looping forever and never actually doing it every `BlockTime`. I think having an inner loop is wasteful, results in busy-looping, and is overall a bad code practice.

Reply:
what about the DA retrieval? DA time can be 10s+ and inner-busy loop can be efficient in that case.

Reply:
Good point. Inside DA retrieval, we call `processNextDABlock` which does a bunch of retries as well: https://github.com/rollkit/rollkit/blob/7b0ec9eb21f8a3bff6daa617eed7e009cff10174/block/manager.go#L510

I still don't think a busy-loop is a good idea since it can run millions of times trying to check the DA layer. In general, I think we should be writing code that avoids busy-looping at all costs. I think having a single outer loop can still be fine for DA retrieval, although if there's a better way than busy-looping, we can consider that too 😃

Reply:
yes the 10 retries are 100ms apart, hence the whole retry spans only 1s. If you were to busy-loop, for a DA blocktime of 10s, the max retries would be <100. The advantage would be that, you could have a guaranteed retrieval time of 100ms instead of 9s in the worst case scenario without busy loop. the choice really depends on how responsive you want the DA retrieval be.

Reply:
Ah I see it doesn't infinite loop for this case. I still think there's no point checking the DA layer every 100ms instead of every (10s) DA block time. Since we'll have fast block sync after this PR is merged, making DA retrieval responsive becomes less of a priority and so I'm fine with avoiding busy-loops here. 

@MSevey @tzdybal Could you please weigh in on the discussion here?

Reply:
If I understand the full flow correctly it is that:
* The `retrieveCond` is signaling every DA block interval and when a block has been produce via `case blockEvent := <-m.blockInCh:` from the `SyncLoop`
* `RetrieveLoop` is a simple infinte loop that is watching for these events
* `processNextDABlock` will retry a few times to retrieve the block. 

If that isn't correct, or is missing some context please lmk.

But based on that I would agree with Manav that an additional loop in `RetrieveLoop` feels unnecessary because of the retires in `processNextDABlock`. If we feel like those retries are not sufficient, then I'd make updates there to keep `RetrieveLoop` simple. 

Additionally, if there are other trigger events, we can add those without a for loop like 

```
// RetrieveLoop is responsible for interacting with DA layer.
func (m *Manager) RetrieveLoop(ctx context.Context) {
        ...
	for {
		select {		
                 case <-ctx.Done():
			return
		case <-waitCh:
		case other:
                }
		daHeight := atomic.LoadUint64(&m.daHeight)
		err := m.processNextDABlock(ctx)
		if err != nil {
			m.logger.Error("failed to retrieve block from DALC", "daHeight", daHeight, "errors", err.Error())
			continue
		}
		atomic.AddUint64(&m.daHeight, 1)
	}
}
```
this would work well if we want to add another check every half a DA interval for example.